### PR TITLE
Implement suggestions of issue #308

### PR DESCRIPTION
**Pull request by Keng12**
Fix issue #308 as suggested by @sloretz

### CODE REVIEW

**Review by sloretz — DISMISSED** [review]
Thanks for the PR @Keng12!

Would you be willing to signoff the commits on this PR to satisfy the DCO bot?

With your branch checked out, this command will signoff the existing commit.

`git rebase HEAD~1 --signoff`

For additional commits, the sign off line can be added `-s` option when committing.

```
git commit -s
```

File: launch_testing_ros/launch_testing_ros/wait_for_topics.py [hunk 1]
```diff
@@ -69,19 +68,14 @@ def _prepare_ros_node(self):
...
-        while self.__running:
-            self.__ros_executor.spin_once(1.0)
-
     def wait(self):
         self.__ros_node.start_subscribers(self.topic_tuples)
         return self.__ros_node.msg_event_object.wait(self.timeout)
 
     def shutdown(self):
-        self.__running = False
+        rclpy.shutdown(context=self.__ros_context)
```
[the remark this answers is not in the document]
Comment (Keng12) [reply launch_testing_ros/launch_testing_ros/wait_for_topics.py]: Hey,
thanks for the suggestion. I changed the line and added a comment.

**Comment by Keng12** [comment]
Thanks for the hint! I signed off my commit this time.

**Comment by Keng12** [comment]
I fixed the signing and commit history.

**Review by jacobperron — APPROVED** [review]
LGTM
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16621/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11205/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17017/)

**Pull request merged**